### PR TITLE
refactor: do not pass contextPage data as props, instead useAnalyticsContext

### PR DESCRIPTION
**Pull request by eessex**
Related to #https://github.com/artsy/force/pull/6482

Remove instances of `contextPage` data passed down to components and implement `useAnalyticsContext` instead.
Ensures that `contextPage` data will be consistent throughout the app in child components.

### CODE REVIEW

File: src/v2/Components/ArtistSeriesRail/ArtistSeriesRail.tsx [hunk 1]
```diff
@@ -63,7 +50,7 @@ const ArtistSeriesRail: React.FC<Props> = props => {
 }
 
 export const ArtistSeriesRailFragmentContainer = createFragmentContainer(
-  ArtistSeriesRail,
+  ArtistSeriesRail as React.FC<Props>,
```
Comment (damassi) [inline src/v2/Components/ArtistSeriesRail/ArtistSeriesRail.tsx:53]: Is this needed?

**Review by damassi — APPROVED** [review]
One minor q, otherwise LGTM 👍

File: src/v2/Components/ArtistSeriesRail/ArtistSeriesRail.tsx (hunk 1 above)
Comment (eessex) [reply src/v2/Components/ArtistSeriesRail/ArtistSeriesRail.tsx:53]: We are still not getting proper typings on relay components, so to have typings we do.

Comment (damassi) [reply src/v2/Components/ArtistSeriesRail/ArtistSeriesRail.tsx:53]: Oh damn, I thought we fixed that :/

**Pull request merged**